log/log.go

```diff
@@ -1,72 +1,185 @@
 package log
 
 import (
-	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"time"
 
-	"github.com/golang/glog"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
 )
 
+type Format string
+
 const (
-	debug glog.Level = glog.Level(4)
-	trace glog.Level = glog.Level(5)
+	FallbackLogFormat       = "text"
+	FallbackTimeFieldFormat = time.RFC3339
 )
 
+var supportedTimeFormats = [...]string{
+	time.ANSIC,
+	time.UnixDate,
+	time.RubyDate,
+	time.RFC822,
+	time.RFC822Z,
+	time.RFC850,
+	time.RFC1123,
+	time.RFC1123Z,
+	time.RFC3339,
+	time.RFC3339Nano,
+	time.Kitchen,
+	time.Stamp,
+	time.StampMilli,
+	time.StampMicro,
+	time.StampNano,
+}
+
+// Configures the global log level and log format.
+func InitializeLogger() zerolog.Logger {
+	zerolog.TimeFieldFormat = resolveTimeFormatFromEnv()
+
+	logSamplerRateAsString, isSamplerRateDefined := os.LookupEnv("LOG_SAMPLER_RATE")
+	if isSamplerRateDefined {
+		logSamplerRate, err := strconv.Atoi(logSamplerRateAsString)
+		if err != nil {
+			log.Warn().Msgf("Provided sampling rate %s cannot be parsed to int32. "+
+				"No sampling rate will be set. Error: %v", logSamplerRateAsString, err)
+		} else if logSamplerRate != 1 {
+			log.Debug().Msgf("Setting log sample rate to every %dth event", logSamplerRate)
+			log.Logger = log.Sample(&zerolog.BasicSampler{N: uint32(logSamplerRate)})
+		}
+	}
+
+	logFormat := resolveLogFormatFromEnv()
+	if logFormat != "json" {
+		log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stdout, TimeFormat: zerolog.TimeFieldFormat})
```

Comment:
```suggestion
		log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stdout, TimeFormat: zerolog.TimeFieldFormat, NoColor: true})
```

Reply:
This is the final change. Everything else is good and working. Once this change is incorporated, we can merge the three PRs.